### PR TITLE
Update README file to be more detailed

### PR DESCRIPTION
**Pull request by mzjp2**
This PR makes the README file a fair bit more detailed, but still keeping linking to the documentation for most things, to keep things in sync. Two new badges have been added (number of downloads and pypi version), plus a main features and installation section and the Zarr logo.

You can see the preview here: https://github.com/mzjp2/zarr-python/blob/readme-update/README.md

TODO:
* [x] Add unit tests and/or doctests in docstrings
* [x] Add docstrings and API docs for any new/modified user-facing classes and functions
* [x] New/modified features documented in docs/tutorial.rst
* [x] Changes documented in docs/release.rst
* [x] Docs build locally (e.g., run ``tox -e docs``)
* [x] AppVeyor and Travis CI passes
* [x] Test coverage is 100% (Coveralls passes)

### CODE REVIEW

**Comment by mzjp2** [comment]
That sounds good @jrbourbeau -- I'll get started on that. Do we have a Zarr logo? 😅

**Comment by dazzag24** [comment]
Hi Zain,

I think the logo is here:
https://github.com/zarr-developers/community/tree/master/logos

File: README.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+- [**Chunk arrays**](https://zarr.readthedocs.io/en/stable/tutorial.html#chunk-optimizations) along any dimension.
+- [**Compress**](https://zarr.readthedocs.io/en/stable/tutorial.html#compressors) and/or filter chunks using any NumCodecs codec.
+- [**Store arrays**](https://zarr.readthedocs.io/en/stable/tutorial.html#tutorial-storage) in memory, on disk, inside a zip file, on S3, etc...
+- [**Read**](https://zarr.readthedocs.io/en/stable/tutorial.html#reading-and-writing-data) an array [**concurrently**](https://zarr.readthedocs.io/en/stable/tutorial.html#parallel-computing-and-synchronization) from multiple threads or processes.
+- Write to an array concurrently from multiple threads or processes.
+- Organize arrays into hierarchies via [**groups**](https://zarr.readthedocs.io/en/stable/tutorial.html#groups).
+
+## Where to get it
+
+Zarr depends on NumPy. It is generally best to [install NumPy](https://docs.scipy.org/doc/numpy/user/install.html) first using whatever method is most appropriate for you operating system and Python distribution. Other dependencies should be installed automatically if using one of the installation methods below.
```
Comment (jrbourbeau) [inline README.md]: I wonder if these install sections could be shorted and then direct to the [installation docs](https://zarr.readthedocs.io/en/stable/#installation) for more info. Something like:

## Installation

Zarr can be installed using `pip`:

```bash
pip install zarr
```

or `conda`:

```bash
conda install -c conda-forge zarr
```

For more details see the [Installation documentation](https://zarr.readthedocs.io/en/stable/#installation).

Comment (mzjp2) [reply README.md]: Agreed and done! 💯

File: README.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+</tr>
+  <td></td>
+  <td>
+    <a href="https://anaconda.org/anaconda/zarr/">
+    <img src="https://anaconda.org/conda-forge/zarr/badges/version.svg" alt="latest release" />
+    </a>
+</td>
+</tr>
+<tr>
+  <td>Package Status</td>
```
[the remark this answers is not in the document]
Comment (mzjp2) [reply README.md]: Thanks for spotting that, rookie mistake. Should have looked through the `setup.py` file more closely. All fixed now, thanks! 🚀 

I take your point re the current status. Maybe something for @alimanfoo or @jakirkham to weigh in on?

**Comment by mzjp2** [comment]
@alimanfoo / @jakirkham any thoughts on this?

**Comment by jakirkham** [comment]
LGTM. Thanks @mzjp2 😄

**Comment by jrbourbeau** [comment]
Thanks @mzjp2!

**Pull request merged**